code-challenges.test.js

```diff
@@ -10,37 +12,104 @@
 const people = [
   { name: "ford prefect", occupation: "a hitchhiker" },
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
   { name: "arthur dent", occupation: "a radio employee" }
 ]
+
+const heartOfGold = (towel) => {
+  const properPeople = towel.map(person => {
+    const nominalName = person.name.replace(/\b\w/g, char => char.toUpperCase())
+    return { name: nominalName, occupation: person.occupation }
+  })
+
+  const noStringsAttached = properPeople.map(person => `${person.name} is ${person.occupation}.`)
+  return noStringsAttached
+}
+
+console.log(heartOfGold(people))
+
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
 // b) Create the function that makes the test pass.
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
+// input: a mixed data array
+// output: an array of only the REMAINDERS of the numbers when divided by three
+// Pseudo: .filter to extract the numbers from the array, then .map shenanigans to do the math-y bits.
+
 // a) Create a test with an expect statement using the variables provided.
 
+describe('booMath', () => {
+  it('return an array of only the REMAINDERS of the numbers when divided by 3 ', () => {
+    expect(booMath(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(booMath(hodgepodge2)).toEqual([ 2, 1, -1 ])
+  });
+});
+
 const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 // Expected output: [ 2, 0, -1, 0 ]
 const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
+const booMath = (input) => {
+  const onlyNums = input.filter(bruh => typeof bruh === "number")
+  remainders = onlyNums.map(brainHurts => brainHurts % 3)
+  return remainders
+}
 // b) Create the function that makes the test pass.
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
+// input: an array of numbers
+// output:the sum of all the numbers cubed.
+// pseudo: .map to do the cubing, then can use .reduce to add the array together.   
+
 // a) Create a test with an expect statement using the variables provided.
 
+  describe('qbert', () => {
+    it('returns the sum of all the numbers cubed', () => {
+      expect(qbert(cubeAndSum1)).toEqual(99)
+      expect(qbert(cubeAndSum2)).toEqual(1125)
+    });
+  });
+
+// b) Create the function that makes the test pass.
+
 const cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 const cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
-// b) Create the function that makes the test pass.
+const qbert = (zork) => {
+  let mappingStuff = zork.map(bingus => bingus ** 3)
+  let reductron = mappingStuff.reduce((a, b) => a + b, 0)
+  return reductron
+}
+
+// ***It seems like 80% of what we do in javascript is using either .map or .filter on stuff? ***
```

Comment:
Great job on this section! 😅 I have used map quite a few times and if it's the right tool for the job, use it.